From da64749d536277810d9cb4c64568cb63d28d7a0d Mon Sep 17 00:00:00 2001
From: Kevin Lin <kevinptt@chromium.org>
Date: Wed, 29 Mar 2023 17:05:07 +0800
Subject: [PATCH] OnceCallbackHolder: Initialize has_run_

As P0883 mentioned, std::atomic performs no initialization, so we should
initialize has_run_ ourselves.

BUG=None
TEST=CQ

Change-Id: Ia78f1dd6f2196862ebce08940ac1a10b802957c7
---
 base/functional/callback_helpers.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/base/functional/callback_helpers.h b/base/functional/callback_helpers.h
index 8bb763638e..97d181af4c 100644
--- a/base/functional/callback_helpers.h
+++ b/base/functional/callback_helpers.h
@@ -10,6 +10,7 @@
 #ifndef BASE_FUNCTIONAL_CALLBACK_HELPERS_H_
 #define BASE_FUNCTIONAL_CALLBACK_HELPERS_H_
 
+#include <atomic>
 #include <memory>
 #include <ostream>
 #include <type_traits>
@@ -90,7 +91,7 @@ class OnceCallbackHolder final {
   }
 
  private:
-  std::atomic<bool> has_run_;
+  std::atomic<bool> has_run_{false};
   base::OnceCallback<void(Args...)> callback_;
   const bool ignore_extra_runs_;
 };
-- 
2.40.0.348.gf938b09366-goog

